app/controllers/books_controller.rb

```diff
@@ -108,4 +110,13 @@ def require_admin
     flash[:danger] = t "only_admin"
     redirect_to books_path
   end
+
+  def rate
+    if @book.reviews.empty?
+      @rate
+    else
+      sum = @book.reviews.pluck(:rate).sum
+      @rate = sum.to_f / @book.reviews.size
+    end
```

Comment:
```suggestion
    return  if @book.reviews.empty?
    
    sum = @book.reviews.pluck(:rate).sum
    @rate = sum.to_f / @book.reviews.size
```
Như này được không em